cache_flat_mutation_reader.hh

```diff
@@ -1041,6 +1042,20 @@ void cache_flat_mutation_reader::maybe_set_static_row_continuous() {
     }
 }
 
+// Last dummies can exist in a quasi-evicted state, where they are unlinked from LRU,
+// but still alive.
+// But while in this state, they mustn't carry any information (i.e. continuity),
+// due to the "older versions are evicted first" rule of MVCC.
+// Thus, when we make an entry continuous, we must ensure that it isn't an
+// unlinked last dummy.
+inline
+void cache_flat_mutation_reader::set_rows_entry_continuous(rows_entry& e) {
+    e.set_continuous(true);
+    if (!e.is_linked()) [[unlikely]] {
+        _snp->tracker()->touch(e);
+    }
+}
```

Comment:
I know all of these issues were found by a randomized test, but could we had a separate regression test for each one?

Another thing you can do is to keep a list of random seeds that must be run on every run of the randomized test because they reproduced known bugs. The problem with this approach, though, is that you have to trust that the random number algorithm will never change, and I don't see how you can trust this unless you add your own random number generator function, and then rerun your million random tests to find reproducers for each of these bugs :-(

Reply:
Those random seeds would be invalidated by any change to the algorithm, even if made preemption choices deterministic.

Reply:
What we might do is add assertions (perhaps not in release mode) about invariants, so that the failure happens earlier. Thus, a failure that would otherwise not be detected (because we didn't read from the bad cache entry) would still cause a crash.